docs/src/content/docs/guides/sidebar.mdx

```diff
@@ -0,0 +1,461 @@
+
+<SidebarPreview
+	config={[
+		{
+			label: 'Guides',
+			items: [
+				{
+					label: 'Components',
+					link: '/guides/components/',
+					badge: { text: 'New', variant: 'default' },
+				},
+			],
+		},
+	]}
+/>
+
+The `badge` property includes a `text` property (the content to display in the badge) and optionally a `variant` property (to define the type of badge shown).
+
+`badge.text` - a string representing the content to display. (e.g. "New")
+
+`badge.variant` - to override the `default` styling setting, set to one of the following properties for custom styling: `note`, `tip`, `danger`, `caution` or `success`.
+
+```js
+starlight({
+	sidebar: [
+		{
+			label: 'Guides',
+			items: [
+				// A link with a green "New" badge.
+				{
+					label: 'Components',
+					link: '/guides/components/',
+					badge: { text: 'Experimental', variant: 'caution' },
+				},
+			],
+		},
+	],
+});
+```
+
+The above configuration will generate the following sidebar:
+
+<SidebarPreview
+	config={[
+		{
+			label: 'Guides',
+			items: [
+				{
+					label: 'Components',
+					link: '/guides/components/',
+					badge: { text: 'Experimental', variant: 'caution' },
+				},
+			],
+		},
+	]}
+/>
+
+## Autogenerated groups
+
+Any group (or sub-group) can be automatically generated, and will be sorted alphabetically by filename. This is helpful when you do not want to manually enter each sidebar item in a group.
+
+To create an autogenerated group, set `autogenerate` to an object specifying the `directory`. For example, given the following file structure:
+
+<FileTree>
+
+- src/
+  - content/
+    - docs/
+      - guides/
+        - components.md
+        - i18n.md
+        - advanced/
+          - project-structure.md
+
+</FileTree>
+
+And the following configuration:
+
+```js
+starlight({
+	sidebar: [
+		{
+			label: 'Guides',
+			// Autogenerate a group of links for the 'guides' directory.
+			autogenerate: { directory: 'guides' },
+		},
+	],
+});
+```
+
+The following sidebar will be generated:
+
+<SidebarPreview
+	config={[
+		{
+			label: 'Guides',
+			items: [
+				{ label: 'Components', link: '/guides/components/' },
+				{ label: 'Internationalization (i18n)', link: '/guides/i18n/' },
+				{
+					label: 'advanced',
+					items: [
+						{ label: 'Project Structure', link: '/guides/project-structure/' },
+					],
+				},
+			],
+		},
+	]}
+/>
+
+### Customizing autogenerated links in frontmatter
+
+Use the [`sidebar` frontmatter field](/reference/frontmatter/#sidebar) to customize autogenerated links.
+
+This field allows you to set a [custom label](/reference/frontmatter/#label) or add a [badge](/reference/frontmatter/#badge) for a page link, to [hide](/reference/frontmatter/#hidden) a link from the sidebar or to define a [custom order](/reference/frontmatter/#order) for all page links.
+
+```md
+---
+title: My page
+sidebar:
+  # Set a custom label for the link
+  label: Custom sidebar label
+  # Show the link in the sidebar
+  hidden: false
```

Comment:
I’m trying to decide what to do about this one. I don’t love showing this usage as a) it’s not how anyone would ever use this, because `false` is the default and b) it risks suggesting this is _needed_ for the link to show in the sidebar.

Do you think it could be OK to just not show `hidden` in use? Or alternatively we could have _two_ code snippet examples and show `hidden: true` in one that then is not shown in the sidebar preview? We could perhaps even try the code snippets in `<CardGrid />` if we wanted to have them side by side on larger screens? Might look terrible, might not.

Reply:
This one was already bothering me too but I think I tried to focus too much on trying to get a 1:1 example with the description above. I think we can just remove it considering we have a link to the frontmatter `hidden` option the line above.

(I tried playing with an extra example for it with a new code snippet but it felt a bit too much for me for such a basic and explicit option.)